### PR TITLE
dmtx-utils: migrate from core

### PR DESCRIPTION
**Pull request by ilovezfs**
Goes together with https://github.com/Homebrew/homebrew-core/pull/8740.

Created with `brew boneyard-formula-pr` because it's marked as abandoned on SourceForge and because upstream has not responded to the report of Imagemagick 7 problems for over 6 months.

### CODE REVIEW

**Comment by swvanderlaan** [comment]
Shoot. I see this is basically abandoned, because apparently `libdmtx` is abandoned on SourceForge, but... It seems the fix is really easy, see: [https://github.com/dmtx/dmtx-utils/issues/2](https://github.com/dmtx/dmtx-utils/issues/2) and the little bit of code changed. My guess is that when the `homebrew-boneyard/dmtx-utils.rb` points to the git page of `dmtx-utils` it will basically be solved. But I could be mistaken - I'm not a coder by study.

So the new code would be something like this?

```
class DmtxUtils < Formula
  desc "Read and write data matrix barcodes"
  **homepage "https://github.com/dmtx/dmtx-utils"**
  **url "https://github.com/dmtx/dmtx-utils.git"**
  **sha256 "DON'T KNOW"
  revision 3**

  depends_on "pkg-config" => :build
  depends_on "libdmtx"
  depends_on "imagemagick"

  def install
    **git clone https://github.com/dmtx/dmtx-utils.git**
    system "./configure", "--disable-dependency-tracking", "--prefix=#{prefix}"
    system "make"
    system "make", "install"
  end
end
```

**Comment by msva** [comment]
By the way, guys, yes, dmtx on SF is abandonned, and I adopted (with permission from original author) it on github. So, currently, I'm a maintainer of dmtx-things, and it's current home is GitHub.

So, not that I have any interest in homebrew (I've no Mac 😎 ), but take the above thing into account, please, and, I guess, reopen the PR :)

**Comment by ilovezfs** [comment]
@msva does it work with imagemagick 7

**Comment by msva** [comment]
Depends.
I didn't specifically fixed it (if there is incompatibilities in API), but I changed link target of `Wand` to `MagicWand` (as it seems to be removed in 7).

And, unfortunatelly, I've no possibility to test it myself, since imagemagick-7 has some compatibility bugs with another 3party soft on my system (so, I personally use 6.9.7.6). But, AFAIRC, @swvanderlaan in mentioned issue on @dmtx tracker mentioned, that "now it works fine" after I fixed link target.

**Comment by ilovezfs** [comment]
@msva thanks for the detailed info. I think we can probably re-add this to Homebrew core.

**Comment by swvanderlaan** [comment]
@ilovezfs that would be awesome. But: it would have to be based on this GitHub as the SourceForge page is still abandoned. In fact, I think it's even better to also link the 'libdmtx' formula to the GitHub-version. Is that possible? Am not familiar with editing formulas, I do have a bioinformatics student working on doing just that for this issue. But any pointers would be much appreciated. Plus: we are happy to test it using our workflow/pipeline as we required the three libraries/packages (imagemagick, libdmtx, and dmtx-utils).

**Comment by swvanderlaan** [comment]
And yes I can confirm that building 'dmtx-utils' from source from GitHub works like a charm.

**Comment by ilovezfs** [comment]
@msva we'll need a tag to build from since we don't have HEAD-only formulae in homebrew/core. Do you mind creating a release tag?

**Comment by msva** [comment]
@ilovezfs yeah, I already suggested it to @swvanderlaan in @dmtx tracker.

Ok. I'll tag new release in few minutes.

**Comment by ilovezfs** [comment]
thanks

**Comment by msva** [comment]
Done. `v0.7.5`

**Comment by ilovezfs** [comment]
Thanks @msva

**Comment by ilovezfs** [comment]
@msva OK so initially:
```
In file included from dmtxread.c:25:
./dmtxread.h:36:10: fatal error: 'wand/magick-wand.h' file not found
#include <wand/magick-wand.h>
         ^
In file included from dmtxread.c:25:
./dmtxread.h:36:10: fatal error: 'wand/magick-wand.h' file not found
#include <wand/magick-wand.h>
         ^
```

Then I replaced in dmtxread/dmtxread.h

```
//#include <wand/magick-wand.h>
#include <ImageMagick-7/MagickWand/MagickWand.h>
```

and now I get

```
Undefined symbols for architecture x86_64:
  "_MagickGetImagePixels", referenced from:
      _main in dmtxread-dmtxread.o
ld: symbol(s) not found for architecture x86_64
Undefined symbols for architecture x86_64:
  "_MagickGetImagePixels", referenced from:
      _main in dmtxread_debug-dmtxread.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [dmtxread.debug] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [dmtxread] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```

**Comment by msva** [comment]
> fatal error: 'wand/magick-wand.h' file not found
> #include <wand/magick-wand.h>

Ok, it seems, they relally heavily moved things around in imagemagick-7 
release, so I'll try to find a way to get a chroot with imagemagick-7 to test 
builds (although, I just checked ubuntus, debians, gentoo, and so on, and they 
all seems to use 6.9 release still (Gentoo has masked 7.0 because of some 
missing headers and problems like with dmtx-utils).

But ok, I'll try to get something like archlinux in the chroot. Looks like 
they already migrated on 7.0

> MagickGetImagePixels

It is strange: that function is in it's place in imagemagick's git HEAD (on GH 
mirror) and was there for more than 5 years.

So, it seems, like MagickWand/MagickWand.h is not a proper replacement.

So, ok, I'll file that issue in @dmtx tracker and will try to fix that ASAP.



// Btw, doesn't homebrew have some kind of "slots" (to allow packages of same 
name, but different versions to co-exist)? Isn't it possible to set 
imagemagick-6.9 as maximum version of dependency for dmtx-utils for now (until 
I fix compatibility with 7)?

**Comment by swvanderlaan** [comment]
Wow. Thanks a lot for your work @ilovezfs and @msva. And yes, the idea of maxing out the version of `ImageMagick` would be a great option for our pipeline - we really don't need 7. But I can imagine that other pipelines that use the same libraries/packages in addition to others _do_ need `ImageMagick` version 7.

**Comment by msva** [comment]
Uhm... it seems, it is not as easy as I assumed to get imagemagick7 even in 
archlinux (it also using 6.9 by default, and it only has 7 as source-only 
package in user repo).

Is it any GNU/Linux distro with imagemagick7 out of the box? CentOS?

// I really have no Macs to test with homebrew on it.

**Comment by swvanderlaan** [comment]
@msva I do have Macs to test it with `homebrew`. No problem. Tell me what to do, and I'll do it. We also have a cluster based on CentOS7 on which we can install any `ImageMagick` (6.8+, 6.9+ or 7+), so here we can also help. What do you need?

**Comment by tbezemer** [comment]
Hi there! I'm the bioinformatics student @swvanderlaan mentioned before. An update from Holland: 

We noticed that the old installation of dmtx-utils on our MacBooks was still working. Because of the issues with this newer version of imagemagick we therefore opted to use those specific legacy versions of each of our dependencies (as found on our MacBooks), to make it work on CentOS7: The ImageMagick version we used was: 6.9.3-10 Q16 x86_64 2017-02-17.

Because we are working on a computing cluster, we do not have the luxury of installing packages using super user/admin rights. We therefore had to install libdmtx in a non-standard location. We ran into some trouble while installing dmtx-utils, which we want to share with you, as it might help others. 

Dmtx-utils' configure step worked fine (we edited our `LD_LIBRARY_PATH`, `LIBRARY_PATH` and `PKG_CONFIG_PATH` to point to these respective locations in the libdmtx installation directory. At the `make` step however, a compiling error was generated, stating that the dmdx.h header file could not be found. This was apparently caused by the fact that the header files within the libdmtx package are located in a non-standard location (/include). We had to manually set the `-I` flag for the compiler setting using the `CPPFLAGS` environment variable:

```
export CPPFLAGS="-I /path_to/libdmtx/include"
./configure --prefix=/path_to/libdmtx

```
This is a little bit hacky and a non-trivial solution to people who are not Unix experts (one of the cluster's unix experts helped us out here). Maybe we can edit the configure/make steps in such a way that the installation of dmtx-utils will also work on systems where non-standard installation is the only way to go?

In short: our conclusion is that, when users are advised to use an older version of imagemagick, libdmtx and dmtx-utils should install and compile fine. Non-standard installation could perhaps be facilitated a little by editing the configure/make steps accordingly?

**Comment by ilovezfs** [comment]
@msva maybe this will help: https://gist.github.com/marcinwol/6c4a713de517fb2ae89f5dd5be0e0ca4

**Comment by msva** [comment]
> @msva maybe this will help:
> https://gist.github.com/marcinwol/6c4a713de517fb2ae89f5dd5be0e0ca4

I already did the trick with overlayfs over my system and force-updating to 
ImgageMagick7.

But... I have blitz-question: does homebrew okay with CMake, or it is 
preferred that I keep autotools? :)

**Comment by ilovezfs** [comment]
both are supported

**Comment by msva** [comment]
I've fixed compatibility with imagemagick-7 and tested that it works fine.

// (and it was faster to make an overlayfs over my rootfs and just unmask and 
upgrade to im7, than chrooting to ubuntus/fedoras/archlinuxes and so on)

So. Now tag `v0.7.5` builds against ImageMagick7 pretty fine and works just 
fine.

But, just in case, autotools needs pkg-config (and that imagemagick installed 
it's *.pc files in place) to detect version of imagemagick installed.


P.S. I also have plans to release 0.7.6 in near future with adding 
GraphicsMagick support and probably migrating to CMake.

**Comment by ilovezfs** [comment]
@msva great, thanks. Here's what I have:
```
class DmtxUtils < Formula
  desc "Read and write data matrix barcodes"
  homepage "http://www.libdmtx.org"
  url "https://github.com/dmtx/dmtx-utils/archive/v0.7.5.tar.gz"
  sha256 "6fa365cd321609db40800f6b34a2387c0ac56ef4643f6101ac83cb762f5ce9ef"

  depends_on "autoconf" => :build
  depends_on "automake" => :build
  depends_on "pkg-config" => :build
  depends_on "libdmtx"
  depends_on "libtool" => :run
  depends_on "imagemagick"

  def install
    system "autoreconf", "-fiv"
    system "./configure", "--disable-dependency-tracking", "--prefix=#{prefix}"
    system "make", "install"
  end

  test do
    system bin/"dmtxread", "--help"
  end
end
```
Some questions:
1. What should the homepage be?
2. Do you have a recommendation for a better test?
3. Are there any changes you'd like to see to the formula?

**Comment by msva** [comment]
> 1. What should the homepage be?

Since libdmtx.org looks like be taken by squatters, I guess, setting github 
repo as homepage will be okay (maybe later I'll make site on github pages for 
dmtx stuff)

> 2. Do you have a recommendation for a better test?

For `dmtxwrite` testing I'd suggest something like `echo "something (or cat 
some file) | dmtxwrite | dmtxread`. Or something that way. Idea is to check 
that text before passing to dmtxwrite is identical to text that dmtxread will 
output. It can also be some files (maybe even binary, just not so big, so 
testing will not eat all the RAM) and you can check not the contents but hash-
sums before and after.

And for `dmtxread` you can use something like
`curl -Ls https://github.com/dmtx/libdmtx/raw/master/test/rotate_test/images/
test_image12.png | dmtxread` (or any other image from https://github.com/dmtx/
libdmtx/test/ )

> 3. Are there any changes you'd like to see to the formula?

For example, I've question about:
>   depends_on "libtool" => :run

Are you sure, that it is not `:build` here?

**Comment by ilovezfs** [comment]
dmtxread and dmtxwrite are ending up linked to libltdl.7.dylib

```
iMac-TMP:~ joe$ brew linkage dmtx-utils
System libraries:
  /usr/lib/libSystem.B.dylib
  /usr/lib/libbz2.1.0.dylib
  /usr/lib/libz.1.dylib
Homebrew libraries:
  /usr/local/opt/freetype/lib/libfreetype.6.dylib (freetype)
  /usr/local/opt/imagemagick/lib/libMagickCore-7.Q16HDRI.2.dylib (imagemagick)
  /usr/local/opt/imagemagick/lib/libMagickWand-7.Q16HDRI.0.dylib (imagemagick)
  /usr/local/opt/libdmtx/lib/libdmtx.0.dylib (libdmtx)
  /usr/local/opt/libtool/lib/libltdl.7.dylib (libtool)
  /usr/local/opt/xz/lib/liblzma.5.dylib (xz)
Possible undeclared dependencies:
  freetype
  xz
iMac-TMP:~ joe$ 
iMac-TMP:~ joe$ brew linkage --reverse dmtx-utils
/usr/lib/libSystem.B.dylib
  bin/dmtxquery
  bin/dmtxread
  bin/dmtxwrite

/usr/lib/libbz2.1.0.dylib
  bin/dmtxread
  bin/dmtxwrite

/usr/lib/libz.1.dylib
  bin/dmtxread
  bin/dmtxwrite

/usr/local/opt/freetype/lib/libfreetype.6.dylib
  bin/dmtxread
  bin/dmtxwrite

/usr/local/opt/imagemagick/lib/libMagickCore-7.Q16HDRI.2.dylib
  bin/dmtxread
  bin/dmtxwrite

/usr/local/opt/imagemagick/lib/libMagickWand-7.Q16HDRI.0.dylib
  bin/dmtxread
  bin/dmtxwrite

/usr/local/opt/libdmtx/lib/libdmtx.0.dylib
  bin/dmtxquery
  bin/dmtxread
  bin/dmtxwrite

/usr/local/opt/libtool/lib/libltdl.7.dylib
  bin/dmtxread
  bin/dmtxwrite

/usr/local/opt/xz/lib/liblzma.5.dylib
  bin/dmtxread
  bin/dmtxwrite
```

which may be because imagemagick is
```
/usr/local/opt/libtool/lib/libltdl.7.dylib
  bin/magick
  lib/ImageMagick/modules-Q16HDRI/coders/aai.so
  lib/ImageMagick/modules-Q16HDRI/coders/art.so
  lib/ImageMagick/modules-Q16HDRI/coders/avs.so
  lib/ImageMagick/modules-Q16HDRI/coders/bgr.so
  lib/ImageMagick/modules-Q16HDRI/coders/bmp.so
  lib/ImageMagick/modules-Q16HDRI/coders/braille.so
  lib/ImageMagick/modules-Q16HDRI/coders/cals.so
  lib/ImageMagick/modules-Q16HDRI/coders/caption.so
  lib/ImageMagick/modules-Q16HDRI/coders/cin.so
  lib/ImageMagick/modules-Q16HDRI/coders/cip.so
  lib/ImageMagick/modules-Q16HDRI/coders/clip.so
  lib/ImageMagick/modules-Q16HDRI/coders/cmyk.so
  lib/ImageMagick/modules-Q16HDRI/coders/cut.so
  lib/ImageMagick/modules-Q16HDRI/coders/dcm.so
  lib/ImageMagick/modules-Q16HDRI/coders/dds.so
  lib/ImageMagick/modules-Q16HDRI/coders/debug.so
  lib/ImageMagick/modules-Q16HDRI/coders/dib.so
  lib/ImageMagick/modules-Q16HDRI/coders/dng.so
  lib/ImageMagick/modules-Q16HDRI/coders/dot.so
  lib/ImageMagick/modules-Q16HDRI/coders/dpx.so
  lib/ImageMagick/modules-Q16HDRI/coders/ept.so
  lib/ImageMagick/modules-Q16HDRI/coders/fax.so
  lib/ImageMagick/modules-Q16HDRI/coders/fd.so
  lib/ImageMagick/modules-Q16HDRI/coders/fits.so
  lib/ImageMagick/modules-Q16HDRI/coders/gif.so
  lib/ImageMagick/modules-Q16HDRI/coders/gradient.so
  lib/ImageMagick/modules-Q16HDRI/coders/gray.so
  lib/ImageMagick/modules-Q16HDRI/coders/hald.so
  lib/ImageMagick/modules-Q16HDRI/coders/hdr.so
  lib/ImageMagick/modules-Q16HDRI/coders/histogram.so
  lib/ImageMagick/modules-Q16HDRI/coders/hrz.so
  lib/ImageMagick/modules-Q16HDRI/coders/html.so
  lib/ImageMagick/modules-Q16HDRI/coders/icon.so
  lib/ImageMagick/modules-Q16HDRI/coders/info.so
  lib/ImageMagick/modules-Q16HDRI/coders/inline.so
  lib/ImageMagick/modules-Q16HDRI/coders/ipl.so
  lib/ImageMagick/modules-Q16HDRI/coders/jnx.so
  lib/ImageMagick/modules-Q16HDRI/coders/jpeg.so
  lib/ImageMagick/modules-Q16HDRI/coders/json.so
  lib/ImageMagick/modules-Q16HDRI/coders/label.so
  lib/ImageMagick/modules-Q16HDRI/coders/mac.so
  lib/ImageMagick/modules-Q16HDRI/coders/magick.so
  lib/ImageMagick/modules-Q16HDRI/coders/map.so
  lib/ImageMagick/modules-Q16HDRI/coders/mask.so
  lib/ImageMagick/modules-Q16HDRI/coders/mat.so
  lib/ImageMagick/modules-Q16HDRI/coders/matte.so
  lib/ImageMagick/modules-Q16HDRI/coders/meta.so
  lib/ImageMagick/modules-Q16HDRI/coders/miff.so
  lib/ImageMagick/modules-Q16HDRI/coders/mono.so
  lib/ImageMagick/modules-Q16HDRI/coders/mpc.so
  lib/ImageMagick/modules-Q16HDRI/coders/mpeg.so
  lib/ImageMagick/modules-Q16HDRI/coders/mpr.so
  lib/ImageMagick/modules-Q16HDRI/coders/msl.so
  lib/ImageMagick/modules-Q16HDRI/coders/mtv.so
  lib/ImageMagick/modules-Q16HDRI/coders/mvg.so
  lib/ImageMagick/modules-Q16HDRI/coders/null.so
  lib/ImageMagick/modules-Q16HDRI/coders/otb.so
  lib/ImageMagick/modules-Q16HDRI/coders/palm.so
  lib/ImageMagick/modules-Q16HDRI/coders/pango.so
  lib/ImageMagick/modules-Q16HDRI/coders/pattern.so
  lib/ImageMagick/modules-Q16HDRI/coders/pcd.so
  lib/ImageMagick/modules-Q16HDRI/coders/pcl.so
  lib/ImageMagick/modules-Q16HDRI/coders/pcx.so
  lib/ImageMagick/modules-Q16HDRI/coders/pdb.so
  lib/ImageMagick/modules-Q16HDRI/coders/pdf.so
  lib/ImageMagick/modules-Q16HDRI/coders/pes.so
  lib/ImageMagick/modules-Q16HDRI/coders/pict.so
  lib/ImageMagick/modules-Q16HDRI/coders/pix.so
  lib/ImageMagick/modules-Q16HDRI/coders/plasma.so
  lib/ImageMagick/modules-Q16HDRI/coders/png.so
  lib/ImageMagick/modules-Q16HDRI/coders/pnm.so
  lib/ImageMagick/modules-Q16HDRI/coders/ps.so
  lib/ImageMagick/modules-Q16HDRI/coders/ps2.so
  lib/ImageMagick/modules-Q16HDRI/coders/ps3.so
  lib/ImageMagick/modules-Q16HDRI/coders/psd.so
  lib/ImageMagick/modules-Q16HDRI/coders/pwp.so
  lib/ImageMagick/modules-Q16HDRI/coders/raw.so
  lib/ImageMagick/modules-Q16HDRI/coders/rgb.so
  lib/ImageMagick/modules-Q16HDRI/coders/rgf.so
  lib/ImageMagick/modules-Q16HDRI/coders/rla.so
  lib/ImageMagick/modules-Q16HDRI/coders/rle.so
  lib/ImageMagick/modules-Q16HDRI/coders/scr.so
  lib/ImageMagick/modules-Q16HDRI/coders/sct.so
  lib/ImageMagick/modules-Q16HDRI/coders/sfw.so
  lib/ImageMagick/modules-Q16HDRI/coders/sgi.so
  lib/ImageMagick/modules-Q16HDRI/coders/sixel.so
  lib/ImageMagick/modules-Q16HDRI/coders/stegano.so
  lib/ImageMagick/modules-Q16HDRI/coders/sun.so
  lib/ImageMagick/modules-Q16HDRI/coders/svg.so
  lib/ImageMagick/modules-Q16HDRI/coders/tga.so
  lib/ImageMagick/modules-Q16HDRI/coders/thumbnail.so
  lib/ImageMagick/modules-Q16HDRI/coders/tiff.so
  lib/ImageMagick/modules-Q16HDRI/coders/tile.so
  lib/ImageMagick/modules-Q16HDRI/coders/tim.so
  lib/ImageMagick/modules-Q16HDRI/coders/ttf.so
  lib/ImageMagick/modules-Q16HDRI/coders/txt.so
  lib/ImageMagick/modules-Q16HDRI/coders/uil.so
  lib/ImageMagick/modules-Q16HDRI/coders/url.so
  lib/ImageMagick/modules-Q16HDRI/coders/uyvy.so
  lib/ImageMagick/modules-Q16HDRI/coders/vicar.so
  lib/ImageMagick/modules-Q16HDRI/coders/vid.so
  lib/ImageMagick/modules-Q16HDRI/coders/viff.so
  lib/ImageMagick/modules-Q16HDRI/coders/vips.so
  lib/ImageMagick/modules-Q16HDRI/coders/wbmp.so
  lib/ImageMagick/modules-Q16HDRI/coders/wpg.so
  lib/ImageMagick/modules-Q16HDRI/coders/xbm.so
  lib/ImageMagick/modules-Q16HDRI/coders/xc.so
  lib/ImageMagick/modules-Q16HDRI/coders/xcf.so
  lib/ImageMagick/modules-Q16HDRI/coders/xpm.so
  lib/ImageMagick/modules-Q16HDRI/coders/xps.so
  lib/ImageMagick/modules-Q16HDRI/coders/xtrn.so
  lib/ImageMagick/modules-Q16HDRI/coders/ycbcr.so
  lib/ImageMagick/modules-Q16HDRI/coders/yuv.so
  lib/ImageMagick/modules-Q16HDRI/filters/analyze.so
  lib/libMagick++-7.Q16HDRI.2.dylib
  lib/libMagickCore-7.Q16HDRI.2.dylib
  lib/libMagickWand-7.Q16HDRI.0.dylib
```

**Comment by msva** [comment]
Yeah, it inherits link to ltdl from ImageMagick:

```
$ lddtree dmtxread/dmtxread
dmtxread/dmtxread (interpreter => /lib64/ld-linux-x86-64.so.2)
    libdmtx.so.0 => /usr/lib64/libdmtx.so.0
    libMagickWand-7.Q16HDRI.so.0 => /usr/lib64/libMagickWand-7.Q16HDRI.so.0
        libX11.so.6 => /usr/lib64/libX11.so.6
            libxcb.so.1 => /usr/lib64/libxcb.so.1
                libXau.so.6 => /usr/lib64/libXau.so.6
                libXdmcp.so.6 => /usr/lib64/libXdmcp.so.6
                    libbsd.so.0 => /usr/lib64/libbsd.so.0
                        librt.so.1 => /lib64/librt.so.1
            libdl.so.2 => /lib64/libdl.so.2
        libgcc_s.so.1 => /usr/lib/gcc/x86_64-pc-linux-gnu/5.4.0/libgcc_s.so.1
    libMagickCore-7.Q16HDRI.so.2 => /usr/lib64/libMagickCore-7.Q16HDRI.so.2
        liblcms2.so.2 => /usr/lib64/liblcms2.so.2
        liblqr-1.so.0 => /usr/lib64/liblqr-1.so.0
            libglib-2.0.so.0 => /usr/lib64/libglib-2.0.so.0
                libpcre.so.1 => /lib64/libpcre.so.1
        libfftw3.so.3 => /usr/lib64/libfftw3.so.3
        libfontconfig.so.1 => /usr/lib64/libfontconfig.so.1
            libexpat.so.1 => /usr/lib64/libexpat.so.1
        libfreetype.so.6 => /usr/lib64/libfreetype.so.6
            libpng16.so.16 => /usr/lib64/libpng16.so.16
            libharfbuzz.so.0 => /usr/lib64/libharfbuzz.so.0
            libfreetype.so.6 => !!! circular loop !!!
                libgraphite2.so.3 => /usr/lib64/libgraphite2.so.3
        libXext.so.6 => /usr/lib64/libXext.so.6
        libbz2.so.1 => /lib64/libbz2.so.1
        libz.so.1 => /lib64/libz.so.1
        libltdl.so.7 => /usr/lib64/libltdl.so.7
    libm.so.6 => /lib64/libm.so.6
    libgomp.so.1 => /usr/lib/gcc/x86_64-pc-linux-gnu/5.4.0/libgomp.so.1
    libpthread.so.0 => /lib64/libpthread.so.0
    libc.so.6 => /lib64/libc.so.6

```

So, it doesn't directly linked against it. It only directed linked to `dmtx`, 
`MagickWand-7.Q16HDRI` and `MagickCore-7.Q16HDRI` (and some system stuff).
And `ltdl` is coming from `MagicCore`.

// you can check that with `objdump`/`readelf`.

So, I guess, dependency on `imagemagick` would be enough, since it in turn 
should have runtime dependency on libtool in that case ;)


But anyway, it is just cosmetic thing, as user will anyway have it installed 
as imagemagick dependency ;)

**Comment by ilovezfs** [comment]
Right, the problem is that `test-bot` will try to uninstall dependencies marked `=> :build` before running the test block, which will make it go :boom: if libtool is removed :)

**Comment by swvanderlaan** [comment]
Hi,
So how's the updating going? I was just trying to re-install `dmtx-utils` from the GitHub manually - it worked before. Now I get this:

> ./autogen.sh
> autogen.sh: running autoreconf
> configure.ac:12: error: possibly undefined macro: AC_PROG_LIBTOOL
>       If this token and others are legitimate, please use m4_pattern_allow.
>       See the Autoconf documentation.
> autoreconf: /usr/local/Cellar/autoconf/2.69/bin/autoconf failed with exit status: 1

Any suggestion?

S

**Comment by msva** [comment]
it's strange:
1) it (macro) was there for eternity (since initial commit in git repo)
2) I'm using autotools 2.69 too, and all works pretty fine...

**Comment by swvanderlaan** [comment]
Never mind! It works - probably updates to the Mac that screwed up things. Ok. No issue.

**Comment by ilovezfs** [comment]
I've opened https://github.com/Homebrew/homebrew-core/pull/10693.
